cli/remotebazel/remotebazel.go

```diff
@@ -928,6 +928,21 @@ func Run(ctx context.Context, opts RunOpts, repoConfig *RepoConfig) (int, error)
 	for {
 		inRsp, executeResponse, latestErr = attemptRun(ctx, bbClient, execClient, req)
 
+		// Handle known error conditions.
+		if latestErr != nil {
+			if strings.Contains(latestErr.Error(), "failed to get snapshot") {
+				log.Warnf("The requested snapshot was not found. Failing the request...")
+				return 1, nil
+			} else if strings.Contains(latestErr.Error(), "requested executor ID not found") {
+				log.Warnf("The requested executor ID was not found. Falling back to remote snapshot...")
+				req.ExecProperties = append(req.ExecProperties, &repb.Platform_Property{
+					Name:  "debug-executor-id",
```

Comment:
are we relying on `debug-executor-id` somewhere? fyi, I think `debug-executor-id` recently broke because of the AskForMoreWork mechanism that was added (executors can "steal" any task even if it's setting debug-executor-id) so at this point it's a best-effort mechanism.

Reply:
This PR https://github.com/buildbuddy-io/buildbuddy/pull/9181/files added debug-executor-id to the Run From Snapshot button, which is why I wanted to add this code. I don't want customers to try and use the generated request, and see confusing failures when the executor has been killed, which may happen fairly frequently

Reply:
I changed the behavior to abort if the requested executor does not exist. I realized the fallback behavior might be more confusing